underlay/src/main/java/bio/terra/tanagra/query/bigquery/BQQueryRunner.java

```diff
@@ -592,4 +593,10 @@ public ExportQueryResult run(ExportQueryRequest exportQueryRequest) {
     return new ExportQueryResult(
         exportFileUrlAndFileName.getRight(), exportFileUrlAndFileName.getLeft());
   }
+
+  // Java17 returns Instant with nanoseconds, BigQuery only supports upto microseconds
+  // Hence, use microseconds precision everywhere
+  public static Instant instantNowMicros() {
```

Comment:
Did not find a shared 'util' class and hence put this function here. It is needed in both underlay and service (see comment)

Reply:
I'm not sure we need to replace `Instant.now()` everywhere with this. In the service, I think we only use Instant in filenames, so this shouldn't matter. And since this is a BQ limitation, is it possible to do this truncation only at the point where we pass it to BQ, rather than farther up the stack?

Maybe a good candidate to discuss over a quick GVC?

Reply:
Set up a call for monday. I debated where this needs to be put. 
- adding it only at BQ calls: this is the minimum need. Created a util function so that we dont need to maintain / change it in multiple locations when needed
- adding in in filenames / sql etc: agree, not needed per say. However using it everywhere will avoid any breaking changes wrt sql. As of filenames, no impact in functionality

Reply:
OK, yeah let's talk it out. My thinking is that since this is a BQ limitation, we should only force it when passing it to BQ. In the future, we may support other cloud backends (e.g. ElasticSearch) which may have their own limitations (e.g. no micros, only full seconds) and so we don't want to tie our codebase too closely to BQ.

Reply:
From discussion moving this to BQ specific code only.

Reply:
Tested (in verily-test) that there is no impact to SQL - 
deployed java 11 - created study, cohort, feature
deployed java 17 - created study, cohort, feature; Updated java11 study (created cohort)
deployed java 11 - Updated java17 study (created feature)